homeassistant/components/filesize/__init__.py

```diff
@@ -2,18 +2,39 @@
 
 from __future__ import annotations
 
+import pathlib
+
 from homeassistant.config_entries import ConfigEntry
 from homeassistant.const import CONF_FILE_PATH
 from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
 
 from .const import PLATFORMS
 from .coordinator import FileSizeCoordinator
 
+FileSizeConfigEntry = ConfigEntry[FileSizeCoordinator]
+
+
+def _get_full_path(hass: HomeAssistant, path: str) -> pathlib.Path:
+    """Check if path is valid, allowed and return full path."""
+    get_path = pathlib.Path(path)
+    if not hass.config.is_allowed_path(path):
+        raise ConfigEntryNotReady(f"Filepath {path} is not valid or allowed")
+
+    if not get_path.exists() or not get_path.is_file():
+        raise ConfigEntryNotReady(f"Can not access file {path}")
+
+    return get_path.absolute()
+
 
-async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+async def async_setup_entry(hass: HomeAssistant, entry: FileSizeConfigEntry) -> bool:
     """Set up from a config entry."""
-    coordinator = FileSizeCoordinator(hass, entry.data[CONF_FILE_PATH])
+    path = await hass.async_add_executor_job(
+        _get_full_path, hass, entry.data[CONF_FILE_PATH]
+    )
+    coordinator = FileSizeCoordinator(hass, path)
```

Comment:
This change adds another executor job to startup. The goal of doing the check inside the data update was to avoid two executor jobs

Reply:
Right, I'll do a partial revert in a follow-up 👍

Reply:
https://github.com/home-assistant/core/pull/117029